src/Build/Instance/ProjectPropertyInstance.cs

```diff
@@ -84,8 +86,21 @@ public string EvaluatedValue
         /// Setter assumes caller has protected global properties, if necessary.
         /// </summary>
         [DebuggerBrowsable(DebuggerBrowsableState.Never)]
-        string IProperty.EvaluatedValueEscaped => _escapedValue;
-
+        string IProperty.EvaluatedValueEscaped
```

Comment:
Put breakpoints on other ways to access value to see if/when they're hit.

Run all tests with a Debugger.Launch as well as try building msbuild/roslyn/C++ project.

Once we understand this better, write it down.

Reply:
See if other PR works with unit test

Reply:
Check conditions on:
[ ] - task execution
[ ] - itemgroup in target
[ ] - itemgroup not in target
[ ] - propertygroup in target
[ ] - import statement